DuckDB.NET.Data/DuckDBConnection.cs

```diff
@@ -45,24 +46,28 @@ public override void Close()
         {
             NativeConnection.Dispose();
             duckDBDatabase.Dispose();
+            connectionIsOpen = false;
         }
 
         public override void Open()
         {
-            var result = PlatformIndependentBindings.NativeMethods.DuckDBOpen(inMemory ? null : filename, out duckDBDatabase);
-            if (result.IsSuccess())
-            {
-                result = PlatformIndependentBindings.NativeMethods.DuckDBConnect(duckDBDatabase, out NativeConnection);
+            if (!connectionIsOpen){
```

Comment:
I think it should throw InvalidOperationException if the connection is already open. What do you think? What do other database providers do?

Reply:
[SqlConnection.Open Exceptions](https://docs.microsoft.com/en-us/dotnet/api/system.data.sqlclient.sqlconnection.open?view=dotnet-plat-ext-5.0#exceptions)

Reply:
It seems that Dapper does a preemptive open of the connection for each operation. I am basing the conclusion on the fact that by merely doing this check, Dapper works fine. And since, Dapper works with all the other providers, looks like this is how they handle it too. I just tried it on SqliteConnection and it didn't complain.

Reply:
@travis-leith Open should throw if it is already opened.

Reply:
Could Dapper be checking the state of the connection before deciding to open the connection then?
Is it possible that `DuckDBConnection` is not exposing this state in a way that Dapper expects?
Just trying to explain why Dapper works for other providers.

Reply:
```C#
var connState1 = duckDBConnection.State.ToString();
duckDBConnection.Open();
var connState2 = duckDBConnection.State.ToString();

Console.WriteLine($"Before open state was {connState1}, now it is {connState2}");
```

results in
"Before open state was Closed, now it is Closed"

Reply:
You are right, the `State` property should be updated during  Open\Close

Reply:
I pushed a change that gets most of the way there. But this is where my C# skills fail me. I don;t know how to override a get property.

Reply:
No worries, thanks for the changes. I will publish a new version now on NuGet.